### PR TITLE
chore: convert to async await syntax

### LINKED ISSUE #27: Convert to use async/await
This module's exported `getResponse` function already returns a promise but it should be converted to use modern async/await syntax, and it's exported `resolver` should be converted to async/await.

### PR DESCRIPTION
**Pull request by PedroMiguelSS**
This PR converts this module's both exported `getResponse` and `resolver` functions to async/await syntax. 

It also adds a util file called `p-try-each` that mimics the same [`async/tryEach`](https://caolan.github.io/async/v3/docs.html#tryEach) method.

Closes #27.

### CODE REVIEW

File: src/resolver.js [hunk 1]
```diff
@@ -1,89 +1,73 @@
...
-          if (err) {
-            return callback(err)
-          }
-
-          return callback(null, dirView.render(path, result.value.Links))
-        })
-      }
+const directory = async (ipfs, path, cid) => {
+  try {
+    const res = await findIndexFile(ipfs, path)
```
Comment (vasco-santos) [inline src/resolver.js]: Can you put the comment back "test if it is a website"? I feel it is important here

File: src/utils/p-try-each.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const pTryEeach = async (iterable) => {
```
Comment (vasco-santos) [inline src/utils/p-try-each.js]: Can you create a module for this?

Comment (PedroMiguelSS) [reply src/utils/p-try-each.js]: Yes, I can and I will. Can we merge this as it is and then update it later to use the module that I'll create?

Comment (hugomrdias) [reply src/utils/p-try-each.js]: you have a typo in the fn name `pTryEeach`should be `pTryEach`

Comment (PedroMiguelSS) [reply src/utils/p-try-each.js]: Oops 🤦‍♂️

**Pull request merged**